src/components/uploadPhotos/UploadSlot.tsx

```diff
@@ -0,0 +1,167 @@
+import React, { useRef } from 'react'
+import { Box, Typography } from '@mui/material'
+import { makeStyles } from 'tss-react/mui'
+import UserPicsType from './UploadPhotos'
+
+interface UserPicsType {
+  id: string
+  url: string | null
+}
+
+interface SlotType {
+  id: string
+  bgPic: string | null
+  userPics: UserPicsType[]
+  setUserPics: (newPics: UserPicsType[]) => void
+  setIsDeleteModalOpened(isOpened: boolean): void
+  setChosenId: (chosenId: string) => void
+  setIsPhotoModalOpened: (isPhotoModalOpened: boolean) => void
+  setChosenUrl: (chosenUrl: string) => void
+}
+
+const UploadSlot: React.FC<SlotType> = ({
+  id,
+  bgPic,
+  setUserPics,
+  userPics,
+  setIsDeleteModalOpened,
+  setChosenId,
+  setIsPhotoModalOpened,
+  setChosenUrl,
+}) => {
+  const { classes } = useStyles()
+
+  const fileInputRef = useRef<HTMLInputElement | null>(null)
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0] as File | undefined
+
+    if (file) {
+      const reader = new FileReader()
+      reader.readAsDataURL(file)
+
+      reader.onloadend = () => {
+        const base64data = reader.result
+
+        // Создаем новый объект Image
+        const img = new Image()
+        img.src = base64data as string
+
+        // Ждем, пока изображение загрузится, и получаем его размеры
+        img.onload = () => {
+          const newPic = {
+            id: id,
+            url: base64data as string,
+            width: img.width, // ширина изображения
+            height: img.height, // высота изображения
```

Comment:
Do we really need to store width and height of the image in local storage?

Reply:
Thought it was necessary for correct photo display but seems that it works fine without these two so I deleted these lines